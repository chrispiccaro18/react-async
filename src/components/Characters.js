import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import style from '../main.css';

function Characters({ characters }) {
  const charLis = characters.map(char => {
    const { name, status, species, image } = char;
    const charProps = { name, status, species, image };
    return <Character key={JSON.stringify({ ...charProps })} {...charProps} />;
  });

  return (
    <ul className={style['char']}>
      {charLis}
    </ul>
  );
}

Characters.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
      species: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired
    })
  )
};

export default Characters;
