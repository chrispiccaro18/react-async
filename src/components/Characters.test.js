import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters comp tests', () => {
  it('creates a Characters comp', () => {
    const characters = [
      {
        name: 'Rick Sanchez',
        status: 'Alive',
        species: 'Human',
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
      },
      {
        name: 'Rick Sanchez',
        status: 'Dead',
        species: 'Human',
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
      },
      {
        name: 'Rick Sanchez',
        status: 'Alive',
        species: 'Alien',
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
      }
    ];
    const wrapper = shallow(<Characters characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
