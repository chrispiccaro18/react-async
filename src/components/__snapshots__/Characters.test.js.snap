// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters comp tests creates a Characters comp 1`] = `
<ul>
  <Character
    image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    key="{\\"name\\":\\"Rick Sanchez\\",\\"status\\":\\"Alive\\",\\"species\\":\\"Human\\",\\"image\\":\\"https://rickandmortyapi.com/api/character/avatar/1.jpeg\\"}"
    name="Rick Sanchez"
    species="Human"
    status="Alive"
  />
  <Character
    image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    key="{\\"name\\":\\"Rick Sanchez\\",\\"status\\":\\"Dead\\",\\"species\\":\\"Human\\",\\"image\\":\\"https://rickandmortyapi.com/api/character/avatar/1.jpeg\\"}"
    name="Rick Sanchez"
    species="Human"
    status="Dead"
  />
  <Character
    image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    key="{\\"name\\":\\"Rick Sanchez\\",\\"status\\":\\"Alive\\",\\"species\\":\\"Alien\\",\\"image\\":\\"https://rickandmortyapi.com/api/character/avatar/1.jpeg\\"}"
    name="Rick Sanchez"
    species="Alien"
    status="Alive"
  />
</ul>
`;
